### ISSUE TITLE
stratum not work

### ISSUE THREAD

**Issue by hackmod** [ethereum-mining/ethminer#1718]
**Describe the bug**
stratum does not work (getWork works fine)

**To Reproduce**
1. `ethminer -v 3 --farm-recheck 2000 -U stratum+tcp://foobar.worker0:x@asia.ethash-hub.miningpoolhub.com:20555`

**Expected behavior**
`ethminer 0.17.0-alpha.1` work. but master branch(274fea8427, 3f7c640292) not work

**Screenshots (Optional)**
![image](https://user-images.githubusercontent.com/32324335/49116725-dd599880-f2e1-11e8-9a34-02553e0fd02b.png)

**Desktop (please complete the following information):**
- Operating System: ubuntu
- Hardware: P106-6G
- Ethminer Version: master branch

**Comment by hackmod** [comment]
for `stratum1+tcp` case. `Worker not authorized` disappeared. but does not work.
(`stratum2+tcp` emit `worker not authorized` error)
![image](https://user-images.githubusercontent.com/32324335/49118748-e6019d00-f2e8-11e8-8c3a-635e342f2df6.png)

**Comment by lesjokolat** [comment]
wrong port see thier support page.

**Comment by hackmod** [comment]
as I already mentioned. `ethminer 0.17.0-alpha.1` works just fine.

![image](https://user-images.githubusercontent.com/32324335/49119627-61188280-f2ec-11e8-820d-a558f7f7a873.png)

**Comment by lesjokolat** [comment]
try the correct port (based on pools own page)does it affect things?

**Comment by lesjokolat** [comment]
ok i see the etc coin was using port 20555

the command lines look to be missing -P can you add that and try?

**Comment by lesjokolat** [comment]
can you try v18 alpha i think they made fixes for worker formats

**Comment by StefanOberhumer** [comment]
Bug came with #1693 
@hackmod  As workaround you can use %2e
( `-P stratum2+tcp://username%2eworkername:x@asia.ethash-hub.miningpoolhub.com:20555` )
I'll try to fix asap.

**Comment by StefanOberhumer** [comment]
@hackmod: Could you try latest master - Please refer to the new syntax using "." ( #1723 )

**Comment by AndreaLanfranchi** [comment]
@hackmod we've just merged #1723 which solves the problem.
Please read the description of #1723

**Comment by AndreaLanfranchi** [comment]
@hackmod please also note that the first response from pool after mining.subscribe is totally wrong.
There are two methods within the result member which is not allowed.

```
{
  "id": 1,
  "result": [
    [
      [
        "mining.set_difficulty",
        "deadbeefcafebabefa49040000000000"
      ],
      [
        "mining.notify",
        "deadbeefcafebabefa49040000000000"
      ]
    ],
    "77ff7d55",
    4
  ],
  "error": null
}
```

Method **must** be a member of the root object.

**Comment by AndreaLanfranchi** [comment]
@hackmod are you sure the errors you're getting are not related to the account you're using ? "foobar"

Can you describe better ?
Is "foobar.etc0" your account name ?

If so, according to #1723 you should write
```
$ ~/path/to/ethminer -v 3 --cuda-devices 0 --farm-recheck 18000 -U stratum+tcp://\`foobar.etc0`:x@asia.ethash-hub.miningpoolhub.com:20555
```
Note the backticks and the escaping

**Comment by hackmod** [comment]
my account info is simply removed.

old version working case:
~~~
 i 18:54:52 stratum  {
        "id" : 3,
        "jsonrpc" : "2.0",
        "method" : "mining.authorize",
        "params" :
        [
                "foobar.etc0",
                "x"
        ]
}
~~~
for master branch case.

stratum+tcp://\`foobar.etc0:x\`@asia.ethash-hub.miningpoolhub.com:20555
~~~
 i 18:56:34 ethminer  << {"id":1,"result":[[["mining.set_difficulty","deadbeefcafebabece02040000000000"],["mining.notify","deadbeefcafebabece02040000000000"]],"680081a7",4],"error":null}
 i 18:56:34 ethminer Stratum mode detected: STRATUM
 i 18:56:34 ethminer Subscribed!
 i 18:56:34 ethminer  >> {"id":3,"jsonrpc":"2.0","method":"mining.authorize",
"params":["foobar.etc0:x","X"]} // <== ???
20d53a845a595c8cdaafcb3993a3a119b5576cf","0x00000000ffb34c02420e9948eacd78cf33b059a88ade1ff0614f7f3c303cf3a7",true]}
 i 18:56:34 ethminer Epoch : 233
 i 18:56:34 ethminer Difficulty : 4.30 Gh
 i 18:56:34 ethminer Job: #689841f5… asia.ethash-hub.miningpoolhub.com:20555
 i 18:56:36 ethminer  << {"id":3,"result":true,"error":null}
 i 18:56:36 ethminer Authorized worker foobar.etc0:x <== ???
~~~

**Comment by AndreaLanfranchi** [comment]
Solved by #1723
